esp/esp/program/modules/handlers/volunteermanage.py

```diff
@@ -91,10 +100,67 @@ def volunteering(self, request, tl, one, two, module, extra, prog):
         else:
             form = VolunteerRequestForm(program=prog)
 
-        context['form'] = form
+        context['shift_form'] = form
+        if 'import_request_form' not in context:
+            context['import_request_form'] = VolunteerImportForm()
         context['requests'] = self.program.getVolunteerRequests()
         return render_to_response('program/modules/volunteermanage/main.html', request, context)
 
+    def volunteer_import(self, request, tl, one, two, module, extra, prog):
+        context = {}
+        response = None
+
+        import_mode = 'preview'
+        if 'import_confirm' in request.POST and request.POST['import_confirm'] == 'yes':
+            import_mode = 'save'
+
+        import_form = VolunteerImportForm(request.POST)
+        if not import_form.is_valid():
+            context['import_request_form'] = import_form
+        else:
+            past_program = import_form.cleaned_data['program']
+            start_date = import_form.cleaned_data['start_date']
+
+            #   Figure out timeslot dates
+            new_requests = []
+            prev_timeslots = []
+            prev_requests = past_program.getVolunteerRequests().order_by('timeslot__start')
+            for prev_request in prev_requests:
+                prev_timeslots.append(prev_request.timeslot)
+            time_delta = start_date - prev_timeslots[0].start.date()
+            for i,orig_timeslot in enumerate(prev_timeslots):
+                new_timeslot = Event(
+                    program = self.program,
+                    event_type = orig_timeslot.event_type,
+                    short_description = orig_timeslot.short_description,
+                    description = orig_timeslot.description,
+                    priority = orig_timeslot.priority,
+                    start = orig_timeslot.start + time_delta,
+                    end   = orig_timeslot.end + time_delta,
+                )
+                #   Save the new timeslot and request only if it doesn't duplicate an existing one
+                if import_mode == 'save' and not Event.objects.filter(program=new_timeslot.program, start=new_timeslot.start, end=new_timeslot.end).exists():
+                    new_timeslot.save()
+                new_request = VolunteerRequest()
+                new_request.program = self.program
+                new_request.timeslot = new_timeslot
```

Comment:
If you didn't save the new timeslot, this won't work -- you need to get the existing one and use that here instead, since you'll end up throwing away the new one.  Same below in the `if`, and further below with `new_request`.  (But see above -- might make this moot.)